nomos-da/network/core/src/broadcast/behaviour.rs

```diff
@@ -0,0 +1,116 @@
+use std::collections::VecDeque;
+use std::task::{Context, Poll};
+
+use libp2p::{Multiaddr, PeerId};
+use libp2p::core::Endpoint;
+use libp2p::swarm::{
+    ConnectionDenied, ConnectionId, FromSwarm, NetworkBehaviour, THandler, THandlerInEvent,
+    THandlerOutEvent, ToSwarm,
+};
+
+use subnetworks_assignations::MembershipHandler;
+
+use super::handler::{BroadcastHandler, DaMessage, HandlerEventToBehaviour};
+
+pub type SubnetworkId = u16;
+
+/// Nomos DA BroadcastEvents to be bubble up to logic layers
+pub enum BroadcastEvent {
+    IncomingMessage { peer_id: PeerId, message: DaMessage },
```

Comment:
Looks like we won't need packing after all. Seems that libp2p will take care of it. Will update messages crate if true.

Reply:
> Looks like we won't need packing after all. Seems that libp2p will take care of it. Will update messages crate if true.

It makes sense is like that, Thanks for taking a look at it.